spec/21-MQTT3_FORMAT.md

```diff
@@ -0,0 +1,51 @@
+# MQTT 3 recommendation
+
+Before MQTT 5 there were no extensibility metadata defined for the messages. Thus
+this section is only a set of recommendations on how trace context can be
+implemented, not the specification.
+
+## JSON payload
+
+It is common that MQTT message payload represented as JSON string. In this case
+it is recommended to set `traceparent` and `tracestate` fields as a fields on a
+outermost JSON object.
+
+For example, if original payload looks like following:
+
+``` json
+{
+    "_type": "cmd",
+    "action": "setConfiguration",
+    "configuration": {
+        "_type":"configuration",
+        "settingA": "value A",
+        "settingB": 42
+        }
+}
+```
+
+Payload with trace context configured will look like following:
+
+``` json
+{
+    "_type": "cmd",
+    "action": "setConfiguration",
+    "configuration": {
+        "_type":"configuration",
+        "settingA": "value A",
+        "settingB": 42
+        },
+    "traceparent": "00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01",
+    "tracestate": "congo=BleGNlZWRzIHRohbCBwbGVhc3VyZS4"
+}
+```
+
+## Handling e2e encryption
```

Comment:
From @zteo-google

End-to-end encryption is generally handled at the transport protocol layer so message payloads are typically not "additionally encrypted". I understand the issue here, and it is similar to the binary payload issue in the paragraph below. Again, my recommendation is not to touch payloads and instead to publish to a separate MQTT topic.

Reply:
```suggestion
## Handling end-to-end encryption
```